### PR TITLE
dev-python/google-api-core:  adding grpcio and grpcio-status dependencies #940642

### PR DESCRIPTION
**Pull request by ormorph**
The` google-api-core` package requires the `grpcio` package to function properly.
When executing this code, an error appears:
```
from google.api_core.gapic_v1 import config
```
In this solution, the `grpcio-status` package is added for normal testing of the `google-api-core` package with the `grpcio` module. Also added a fix for the `dev-python/google-auth` dependency in the package, it requires a newer version of the package.

---

Please check all the boxes that apply:

- [x] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [x] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [x] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [x] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

**Comment by ormorph** [comment]
Now I tried synchronization via the web interface for the first time, and I realized that it’s better not to do it that way. I won't do this again, I had to upload the repository again.

**Comment by mgorny** [comment]
There's a bunch of checkboxes in the PR template that you need to confirm, before I can even take look at the code.

**Comment by ormorph** [comment]
What are these checkboxes? Not the ones I installed in the `files changed`? - `Viewed`.  Since I have already created a PR, the template is no longer available to me. Or maybe I misunderstood.

**Comment by mgorny** [comment]
https://github.com/gentoo/gentoo/blob/master/.github/pull_request_template.md

**Comment by ormorph** [comment]
> https://github.com/gentoo/gentoo/blob/master/.github/pull_request_template.md

I don't understand, do I need to put X's in the markdown file?  I did it via web form, but I can do it all in one commit.

**Comment by mgorny** [comment]
No, you are supposed to copy it into your pull request description and fill it afterwards.

**Comment by ormorph** [comment]
Ok, now I'll delete the commit and add this.

**Comment by ormorph** [comment]
Did I do it right this time?

**Comment by mgorny** [comment]
Yes, thank you.

**Comment by ormorph** [comment]
Okay, stop, I'll re-upload the changes now. Initially I added something else. But after these returns it was apparently lost. I'll re-upload the commit now.

**Comment by ormorph** [comment]
While I was adding the patch, it looks like the ports tree was updated, and the google-api-core-2.19.2.ebuild file was deleted, I'll make some corrections and re-upload it now.

**Comment by ormorph** [comment]
Added changes to the test branch for now. Although my changes in the test version ~ I find more stable.

**Comment by ormorph** [comment]
I'll re-upload now, rebase and change the nickname to the first and last name in the commit header. Everything else will remain the same.

**Comment by ormorph** [comment]
It's strange why I have `no signoff` written here? I created a commit with the `-s` parameter. I wonder if this could be a consequence of me disabling gpg checking.
```
git config --global commit.gpgsign false
```

File: dev-python/grpcio-status/grpcio-status-1.66.2.ebuild [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{10..13} )
+DISTUTILS_USE_PEP517=setuptools
+inherit distutils-r1 pypi
+
+DESCRIPTION="Reference package for GRPC Python status proto mapping"
+HOMEPAGE="https://grpc.io"
```
Comment (mgorny) [inline dev-python/grpcio-status/grpcio-status-1.66.2.ebuild:11]: Include GitHub and PyPI URLs for homepages.

File: dev-python/grpcio/grpcio-1.66.2.ebuild [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+DESCRIPTION="High-performance RPC framework (python libraries)"
+HOMEPAGE="https://grpc.io"
+
+LICENSE="Apache-2.0"
+SLOT="0"
+KEYWORDS="~amd64 ~arm64 ~x86"
+
+RDEPEND="
+	dev-libs/openssl[-bindist(-)]
+	dev-libs/re2
```
Comment (mgorny) [inline dev-python/grpcio/grpcio-1.66.2.ebuild:19]: How are they used? (i.e. compiled into extensions, via ctypes or…?)

File: dev-python/grpcio/grpcio-1.66.2.ebuild [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+	dev-python/protobuf-python[${PYTHON_USEDEP}]
+	dev-python/six[${PYTHON_USEDEP}]
+	net-dns/c-ares
+	sys-libs/zlib
+"
+
+BDEPEND="${RDEPEND}
```
Comment (mgorny) [inline dev-python/grpcio/grpcio-1.66.2.ebuild:26]: Are you convinced that all of these are `BDEPEND`? Why would they?

Also Cython suggests you're building a C extension, in which case `DISTUTILS_EXT` would be necessary. And I'm pretty sure the eclass will output a warning about that.

**Review by mgorny — CHANGES_REQUESTED** [review]
A few general remarks so far:

1. One commit per package, use package prefix for commit messages. Either look at the commit history or at the devmanual on how to do things.
2. The newly added packages don't run tests.

File: dev-python/grpcio/grpcio-1.66.2.ebuild [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+PYTHON_COMPAT=( python3_{10..13} )
+DISTUTILS_USE_PEP517=setuptools
+inherit distutils-r1 pypi
+
+DESCRIPTION="High-performance RPC framework (python libraries)"
+HOMEPAGE="https://grpc.io"
+
+LICENSE="Apache-2.0"
+SLOT="0"
+KEYWORDS="~amd64 ~arm64 ~x86"
```
Comment (mgorny) [inline dev-python/grpcio/grpcio-1.66.2.ebuild:15]: Did you actually test it on each of these architectures?

File: dev-python/grpcio/grpcio-1.66.2.ebuild (hunk 3 above)
Comment (ormorph) [reply dev-python/grpcio/grpcio-1.66.2.ebuild:24]: Fixed

File: dev-python/grpcio/grpcio-1.66.2.ebuild (hunk 4 above)
Comment (ormorph) [reply dev-python/grpcio/grpcio-1.66.2.ebuild:18]: > Did you actually test it on each of these architectures?

Only on `amd64` and `arm64`. I can't check on x86, but I can't just delete it, there will be pkgcheck errors.

**Comment by ormorph** [comment]
I added corrections, you can check if I did everything correctly.

File: dev-python/grpcio/grpcio-1.66.2.ebuild (hunk 4 above)
Comment (mgorny) [reply dev-python/grpcio/grpcio-1.66.2.ebuild:18]: Delete it from all packages (including new version of `google-api-core`), and we'll readd it after testing.

File: dev-python/grpcio/grpcio-1.66.2.ebuild (hunk 3 above)
Comment (mgorny) [reply dev-python/grpcio/grpcio-1.66.2.ebuild:24]: Is protobuf-python a BDEPEND?

**Review by mgorny — COMMENTED** [review]
There are still no tests in `grpcio*` packages.

**Comment by ormorph** [comment]
> There are still no tests in `grpcio*` packages.

There are no such tests for the grpcio package. This package is tested in tests of other packages, for example, the same google-api-core.
There is a test for the grpcio-status package [here](https://github.com/grpc/grpc/tree/master/src/python/grpcio_tests/tests/status).  But this test is missing from the archive with the grpcio_status-1.66.2.tar.gz source code. If you start it manually, of course it will pass.

**Comment by ormorph** [comment]
> Is protobuf-python a BDEPEND?

Yes, I missed it, it will be corrected.

**Comment by ormorph** [comment]
It's too early to test anything yet, grpcio will need a major rework. By default, it compiles statically with the libraries that come with the source code. We will need to consider another option for building with system libraries. Then, after checking, it will be necessary to add new flags to profiles/use.local.desc. For now this will hang around for another two days or so. I guess I can still test this on x86 later.

**Comment by ormorph** [comment]
Added changes.  Added a simple patch for static assembly under `x86`, which forces the condition to immediately go to else, so that the `OPENSSL_NO_ASM` macro is added. Otherwise, the x86 build will fail.
All tests pass on all architectures. I thought it would take me longer. Can you check if I did everything correctly?
Linking of system libraries occurs with the file `cygrpc.cpython-312-x86_64-linux-gnu.so`. Depending on the architecture, the name may differ. All tests complete normally.

File: dev-python/grpcio/grpcio-1.66.2.ebuild [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+IUSE="system-cares system-re system-ssl system-zlib"
+
+RDEPEND="
+	dev-python/protobuf-python[${PYTHON_USEDEP}]
+	system-cares? ( net-dns/c-ares )
+	system-re? ( dev-libs/re2 )
+	system-ssl? ( dev-libs/openssl )
+	system-zlib? ( sys-libs/zlib )
```
[the remark this answers is not in the document]
Comment (ormorph) [reply dev-python/grpcio/grpcio-1.66.2.ebuild:26]: So, as I understand it, static assembly is not needed? Or do I need to enable default flags in the ebuild?

[the remark this answers is not in the document]
Comment (mgorny) [reply dev-python/grpcio/grpcio-1.66.2.ebuild:26]: Yes, just force using system libs.

**Comment by ormorph** [comment]
I really hope that I did the right thing this time. But somehow I already have vague doubts about this.

**Comment by mgorny** [comment]
Sorry, looks like reports are borked. I'm going to look at fixing that now. In any case, let's get the packages working first, and then we'll worry about CI.

**Comment by ormorph** [comment]
Just now `pkgcheck scan --commits --net` started working, before that it immediately terminated the work and not a single message about the search.  Before this, only local `pkgcheck scan` worked fine. Now I'm waiting for the search to finish, and I've already updated the Manifest for all packages. I think it should start working further.  It coincided with the fact that the report pages also began to load.

**Comment by ormorph** [comment]
Tests are already in all packages. But the request for changes is hanging. On x86 tests were also carried out.

**Comment by ormorph** [comment]
> You still aren't running the tests from `grpcio_testing`. While I realize many of these tests are irrelevant to the two packages in question, I suspect that there is a subset applicable to them.

The `grpcio_testing` package is not related to these packages in any way. Besides, I have already said that there is no test for the grpcio package and never was. The only test is the import test.  What tests can we talk about from the [grpcio_testing](https://pypi.org/project/grpcio-testing/#files) package? This package does not contain any tests at all.

**Comment by mgorny** [comment]
Upstream talks of testing here: https://github.com/grpc/grpc/blob/master/src/python/CONTRIBUTING.md#steps-to-contributing-to-grpc-python

You can't use bazel but you should be able to run them without it.

**Comment by ormorph** [comment]
> Upstream talks of testing here: https://github.com/grpc/grpc/blob/master/src/python/CONTRIBUTING.md#steps-to-contributing-to-grpc-python
> 
> You can't use bazel but you should be able to run them without it.

To pass this test all of these packages must be installed:
```
grpcio
grpcio_admin
grpcio_channelz
grpcio_csds
grpcio_csm_observability
grpcio_health_checking
grpcio_observability
grpcio_reflection
grpcio_status
grpcio_testing
```
It's easier for me to just close this PR.  If this is what you want, then I will do it. I wouldn't have to edit the ebuilds so many times for this. This is not a test that can be added to ebuilds, as its archive is missing.

**Comment by mgorny** [comment]
I've finally managed some time to start working on it. Good news is, the test suite seems pretty workable. Bad news, the package in general is a terrible piece of shit with more shit bundled, and that bundled shit bundles even more shit, so getting it all right is going to take some time.

**Comment by mgorny** [comment]
Thanks again. I've finished working on it, and I'm about to push it with working tests. I've temporarily removed `~arm64` since I don't have an arm64 machine handy, but it'll be readded shortly.

**Pull request closed without merging**